doc/interfaces/registration_flow.rst

```diff
@@ -0,0 +1,172 @@
+DIPY. You can check how to :ref:`fetch the DIPY data<data_fetch>`.
+
+-------------------
+Affine Registration
+-------------------
+
+Affine registration is an optimization strategy used to maximize the mutual
+information between two volumes or images. The affine transformation allows
+linear coordinate changes that include shearing, scaling, rotation and
+translation.
+
+The workflow for the affine registration requires the paths to the static image
+file, and to the moving image file. Static image is used as a reference and
+moving image is transformed during optimization.
+
+You may want to create an output directory to save the transformed image::
+
+    mkdir affine_reg_output
+
+Run the following command::
+
+    dipy_align_affine <path_to_static_file> <path_to_moving_file> --out_dir "affine_reg_output" --out_affine "affine_reg.txt"
+
+This command will apply affine transformation on the moving image file and save
+the transformed image and the affine matrix to the directory
+``affine_reg_output``.
+
+In case you did not specify the output directory, the transformed image file
+and affine matrix would be saved in the input directory by default. If you did
+not specify the name of the output affine matrix, the affine matrix will be
+saved to a file named ``affine.txt`` by default, located in the output
+directory also by default.
+
+-----------------------
+Image Registration Flow
+-----------------------
+
+Image registration flow is a generic workflow which can be used to perform
+different types of registration such as center of mass, transformation, rigid
+body affine which includes translation, rotation, shearing and scaling. We can
+intend to use only one type of registration. Alternatively, using image
+registration, we can perform a registration in a progressive manner. For
+example, using affine registration with ``progressive`` set to ``True`` will
+involve center of mass, translation, rigid body and full affine registration.
+Whereas, when ``progressive`` is ``False`` the registration will include only
+center of mass and affine registration. The progressive registration will be
+slower but will improve the quality.
+
+We will first create a directory in which to save the transformed image and the
+affine matrix (e.g.: ``image_reg_output``)::
+
+    mkdir image_reg_output
+
+To run the image registration, we need to specify the paths to the static image
+file, and to the moving image file, followed by the optional arguments. In this
+case, we will be specifying the type of registration to be performed
+(``transform``) and the output directory (``out_dir``).
+
+To perform center of mass registration, we will call the ``dipy_align_affine``
+command with ``transform`` parameter set to ``com`` e.g.::
+
+    dipy_align_affine <path_to_static_file> <path_to_moving_file> --transform "com" --out_dir "image_reg_output"
+
+This command will save the transformed image and the affine matrix to the
+specified output directory.
+
```

Comment:
So I'd re-work the affine part here as:

If we are to use an affine transformation type during the registration process, we would call the {double_backticks}dipy_align_affine{double_backticks} command as, e.g.::

    dipy_align_affine <path_to_static_file> <path_to_moving_file> --transform "affine" --out_dir "affine_reg_output" --out_affine "affine_reg.txt"

This command will apply an affine transformation on the moving image file, and save
the transformed image and the affine matrix to the ``affine_reg_output`` directory.

In case you did not specify the output directory, the transformed image file
and affine matrix would be saved to the current by default. If you did
not specify the name of the output affine matrix, the affine matrix will be
saved to a file named ``affine.txt`` by default, located in the current
directory also by default.

Reply:
Addressed in 2b43d64